### PR TITLE
feat(keycloak-backend)!: migrate to the new backend and remove deprecations

### PR DESCRIPTION
**Pull request by dzemanov**
## General
- removes legacy backend system
- removes deprecations

## Fixes
[RHIDP-4056](https://issues.redhat.com/browse/RHIDP-4056)

### CODE REVIEW

File: plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts [hunk 1]
```diff
@@ -123,34 +126,44 @@ export class KeycloakOrgEntityProvider implements EntityProvider {
...
 
   static fromConfig(
-    configRoot: Config,
-    options: KeycloakOrgEntityProviderOptions,
+    deps: {
+      config: Config;
+      logger: LoggerService;
+    },
+    options: (
+      | { schedule: 'manual' | SchedulerServiceTaskRunner }
```
Comment (schultzp2020) [inline plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts:134]:
```suggestion
      | { SchedulerServiceTaskRunner }
```

I think we should require a schedule.

File: plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts [hunk 2]
```diff
@@ -123,34 +126,44 @@ export class KeycloakOrgEntityProvider implements EntityProvider {
...
-        logger: options.logger,
+        logger: logger,
         userTransformer: options.userTransformer,
         groupTransformer: options.groupTransformer,
       });
 
-      if (taskRunner !== 'manual') {
+      if (typeof taskRunner !== 'string') {
         provider.schedule(taskRunner);
       }
```
Comment (schultzp2020) [inline plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.ts:169]:
```suggestion
```

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

File: plugins/keycloak-backend/src/module/catalogModuleKeycloakEntityProvider.ts [hunk 3]
```diff
@@ -59,13 +60,14 @@ export const catalogModuleKeycloakEntityProvider = createBackendModule({
...
-            logger,
-            scheduler,
-            userTransformer: userTransformer,
-            groupTransformer: groupTransformer,
-          }),
+          KeycloakOrgEntityProvider.fromConfig(
+            { config, logger },
+            {
+              scheduler: scheduler,
+              userTransformer: userTransformer,
```
Comment (dzemanov) [inline plugins/keycloak-backend/src/module/catalogModuleKeycloakEntityProvider.ts:67]: Do we want to add here default schedule @schultzp2020?  And update docs accordingly.

```suggestion
              scheduler: scheduler,
              schedule: scheduler.createScheduledTaskRunner({
                frequency: { minutes: 30 },
                timeout: { minutes: 3 },
              }),
              userTransformer: userTransformer,
```

Comment (schultzp2020) [reply plugins/keycloak-backend/src/module/catalogModuleKeycloakEntityProvider.ts:67]: Yes

Comment (schultzp2020) [reply plugins/keycloak-backend/src/module/catalogModuleKeycloakEntityProvider.ts:67]: Yes

File: plugins/keycloak-backend/src/index.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@
 export * from './providers';
 export type { UserTransformer, GroupTransformer } from './lib';
 export * from './lib/transformers';
-export * from './dynamic/index';
 export * from './extensions';
 export * from './lib/types';
+export * from './module';
+export { default } from './module';
```
Comment (schultzp2020) [inline plugins/keycloak-backend/src/index.ts:23]:
```suggestion
export { catalogModuleKeycloakEntityProvider as default } from './module';
```

File: plugins/keycloak-backend/src/index.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@
 export * from './providers';
 export type { UserTransformer, GroupTransformer } from './lib';
 export * from './lib/transformers';
-export * from './dynamic/index';
 export * from './extensions';
 export * from './lib/types';
+export * from './module';
```
Comment (schultzp2020) [inline plugins/keycloak-backend/src/index.ts:22]:
```suggestion
```

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**